src/pykeen/datasets/base.py

```diff
@@ -431,9 +430,9 @@ class RemoteDataset(PathDataset):
     def __init__(
         self,
         url: str,
-        relative_training_path: str,
-        relative_testing_path: str,
-        relative_validation_path: str,
+        relative_training_path: pathlib.Path,
```

Comment:
uhhh this one doesn't make sense, since you're giving a path part right? At least if it's possible to give a path part in a pathlib.Path, you need to make sure this can accept either.

Reply:
In all subclasses I did use [PurePath](https://docs.python.org/3/library/pathlib.html#pure-paths). Otherwise, I might change them to `Sequence[str]`, so you can give the individual parts?

Reply:
Otherwise you will end up with OS specific hard-coded paths, likely not working under Windows(?)

Reply:
I checked the documentation again, and it seems like PurePath is also the most generic type: 

<img src="https://docs.python.org/3/_images/pathlib-inheritance.png"/>